memo/_base.py

```diff
@@ -1,6 +1,78 @@
 import orjson
-from typing import Callable, List
+from typing import Callable, Dict, List, Iterable
+from types import GeneratorType
 from functools import wraps
+from joblib import Parallel, delayed, parallel_backend
+import joblib.parallel
+from rich.progress import Progress
+import time
+import warnings
+
+
+class Runner():
+
+    def __init__(self, *args, **kwargs, ):
```

Comment:
It seems fine to have keyword arguments around so that we can add extra parameters to the underlying compute engine, but at the very least we should have a parameter for `"backend"`, no?

Reply:
I also imagine we'd always want a `n_jobs` parameter as well now that I think about it. 

Got an opinion on default values for `backend` and `n_jobs`?

Reply:
I can see the argument for explicitly using `n_jobs` and `backend` keywords. Easy fix to put them in. As for defaults I've always found for notebook based work on my local machine that "threading" usually works the best and I default n_jobs to -1 however I don't know what joblib does under the hood with that combination since you can have many more threads than cpu cores.  With that said though I think it would be best to leave the defaults as `None`  for `n_jobs` and  `loky` for `backend` since those are the defaults used in joblib. I think it will be easy enough for the user to override them.

Reply:
`None` for `n_jobs` makes sense. I'm wondering about `"loky"` for `backend` though. Mainly because locally I didn't see it get a speedup. I might need to set the `n_jobs` to 2 instead of -1 there to see the speedup.

Reply:
Pushed the code with n_jobs defaulted to None for now.